# M0: LARRY data reproduction
Official loader and Quickstart clone/type subset. Seed 0. Source: https://www.scdiffeq.com/_tutorials/quickstart.html

In [1]:
from pathlib import Path
import os,sys,json
ROOT=Path.cwd()
if ROOT.name=='notebooks':ROOT=ROOT.parent
os.chdir(ROOT);sys.path.insert(0,str(ROOT))


In [2]:
from src.preflight import inspect_environment
status=inspect_environment(Path('data'))
print(status['python'],status['packages'])

3.11.17 {'torch': '2.6.0+cu124', 'scdiffeq': '1.1.4', 'torchsde': '0.2.6', 'anndata': '0.12.19', 'scanpy': '1.11.5', 'numpy': '2.4.6'}


In [3]:
import lightning as pl,scdiffeq as sdq
from src.data import quickstart_subset,save_inventory
pl.seed_everything(0,workers=True)
adata_ref=sdq.datasets.larry(data_dir='data',variant=None)
adata=quickstart_subset(adata_ref)
summary=save_inventory(adata_ref,adata,Path('outputs'))
summary

Seed set to 0


scDiffEq [INFO]: Loading data from data/scdiffeq_data/larry/larry.processed.h5ad


{'full': {'n_cells': 130887,
  'n_genes': 2492,
  'pca_shape': [130887, 50],
  'keys': {'obs': ['Library',
    'Cell barcode',
    'Time point',
    'Starting population',
    'Cell type annotation',
    'Well',
    'SPRING-x',
    'SPRING-y',
    'clone_idx',
    'fate_observed',
    't0_fated',
    'train',
    'ct_score',
    'ct_pseudotime',
    'ct_num_exp_genes'],
   'obsm': ['X_clone', 'X_pca', 'X_scaled', 'X_umap', 'cell_fate_df'],
   'layers': ['X_scaled'],
   'uns': ['fate_counts', 'h5ad_path', 'time_occupance']}},
 'quickstart_subset': {'n_cells': 9350,
  'n_genes': 2492,
  'pca_shape': [9350, 50],
  'keys': {'obs': ['Library',
    'Cell barcode',
    'Time point',
    'Starting population',
    'Cell type annotation',
    'Well',
    'SPRING-x',
    'SPRING-y',
    'clone_idx',
    'fate_observed',
    't0_fated',
    'train',
    'ct_score',
    'ct_pseudotime',
    'ct_num_exp_genes',
    'source_cell_id',
    'nm_clones'],
   'obsm': ['X_pca', 'X_scaled', 'X_umap'],
   '

In [4]:
import pandas as pd
from IPython.display import display
display(pd.read_csv('outputs/tables/dataset_summary.csv'))
display(pd.read_csv('outputs/tables/quickstart_subset_time_by_class.csv'))

,dataset,n_cells,n_genes
0,full,130887,2492
1,quickstart_subset,9350,2492


,Time point,Cell type annotation,cell_count
0,2.0,Undifferentiated,249
1,4.0,Monocyte,719
2,4.0,Neutrophil,714
3,4.0,Undifferentiated,1633
4,6.0,Monocyte,2173
5,6.0,Neutrophil,2386
6,6.0,Undifferentiated,1476


Full: 130887 cells × 2492 genes. Quickstart subset: 9350 cells × 2492 genes, 50 PCs. Official cached PCA/scaling may use all cells; this reproduction does not establish independent preprocessing. Provenance and hashes: `outputs/logs/data_provenance.json`.

In [5]:
provenance=json.loads(Path('outputs/logs/data_provenance.json').read_text())
print([(p['path'],p['size'],p['sha256']) for p in provenance['files']])
del adata_ref,adata

[('data/scdiffeq_data/larry/_larry.raw.h5ad', 5308287542, 'a2fdec7c7b0022816546a6744d8d4ab9e4bdc60d6fdb5ade8902b463d6c06919'), ('data/scdiffeq_data/larry/larry.ct_obs_df.csv', 6373034, '0eac896e4abb67dc24d96abab5b4aa36258fc52fa8f8b630a971cf7477e461b2'), ('data/scdiffeq_data/larry/larry.ct_var_df.csv', 82171, 'b6626d62ea1370232801065a53e298cfedfdaa787d159c1b8b9b469c104ef772'), ('data/scdiffeq_data/larry/larry.processed.h5ad', 7920816770, '23af3ad0591496f370f036d23166700dcc26ec2144ba1fd88be10f8c92a4c224'), ('data/scdiffeq_data/larry/pca.pkl', 1018691, '396c6c6eb84a4c778769bf3d7d4888aa284248a53fa11776fe6f91b2254b2f12'), ('data/scdiffeq_data/larry/scaler.pkl', 60220, 'fdff63390026768e99bcc2bbb3f7c987b6ff00ba4b11da13caa16374355e76f3')]
